test/grouping.jl

```diff
@@ -118,112 +116,118 @@ module TestGrouping
         f7(df) = (c2 = df[:c].^2,)
         f8(df) = DataFrame(c2 = df[:c].^2)
 
-        res = unique(df[cols])
-        res.cmax = [maximum(df[(df.a .== a) .& (df.b .== b), :c])
+        for cols in ([:a, :b], [:b, :a], [1, 2], [2, 1], [true, true, false])
```

Comment:
it would be good to add a test also when grouping columns are e.g. column 5 and 3 or something similar (not two first columns). I have checked it manually and it is now OK, but I did not find this covered in tests.

Reply:
Indeed. I've added a commit. These tests are becoming more complex than the actual code, but I guess there's no way around that. ;-)

Reply:
Thanks. Indeed we probably cannot avoid it, but it is excellent to have them.